### PR TITLE
Fix multiline_text's overriding of set_cursor

### PR DESCRIPTION
**Pull request by stevecotton**
When code in text_box_base.cpp calls the virtual method, it will only consider the 2-argument version, and thus skip the 3-argument wrapper. Override the 2-argument method to catch the calls and have the intended extra effect.

Note: in testing I don't see a difference in what happens in the UI, but I'm not familiar with the edge cases of this widget.
I just checked the stack traces to confirm that the call from text_box_base::insert_char skipped the wrapper.

This fixes a Clang warning, although the warning itself points out the opposite effect - that the 3-argument version hid the 2-argument one in the subclass.

Also fix a Clang warning that scroll_text could mark a method as an override.

### CODE REVIEW

File: src/gui/widgets/multiline_text.hpp
Comment (babaissarkar) [inline src/gui/widgets/multiline_text.hpp:142]: I rewrote the scrolling code in the last few commits, and it seems that the autoscroll check is not needed anymore. So you can remove the that check altogether and make it 2 argument. (also remove the corresponding method call from here : https://github.com/wesnoth/wesnoth/blob/8d3a52eb3fb4e32c5e013eac0bf0a5d7c63839ec/src/gui/widgets/multiline_text.cpp#L215) and remove the comment as well.

Comment (babaissarkar) [reply src/gui/widgets/multiline_text.hpp:142]: To test it, just paste some big text and try to navigate and type something, just like in any text editor.

Comment (stevecotton) [reply src/gui/widgets/multiline_text.hpp:142]: My preference in that case would be to close this PR, and wait for one from you.

**Comment by stevecotton** [comment]
I think this (and possibly a warning about deprecated SSL methods) are the only warnings that appear with Clang. If so, I'm looking to fix it before release as a nice-to-have, because it'll avoid getting a "fails to build from source with Clang" bug later on in Debian or Ubuntu.

**Comment by babaissarkar** [comment]
Is it possible for me to directly edit the files in this PR?

**Comment by stevecotton** [comment]
Since you got added as a project member, I think so - the PR's "allow edits by maintainers" is on. If you add my fork as a remote repo then you should be able to push and force-push to this branch of it.

**Comment by babaissarkar** [comment]
> BTW, for Wesnoth commits that I'm author of, feel free to take my changes into a commit that lists yourself as the author - I feel that my name's already associated with a lot of files, so it's no problem to mix up the authorship. However, don't do that with @CelticMinstrel 's commits, as he feels the opposite (and for anyone else you'd have to ask).

Not going to edit anyone else's PRs, don't worry. I just did it for this one because it was much better than creating another PR, and I'd also asked for your permission before pushing here.

File: src/gui/widgets/multiline_text.hpp
Comment (stevecotton) [inline src/gui/widgets/multiline_text.hpp:135]: This "defaults the autoscroll argument" is outdated

Comment (babaissarkar) [reply src/gui/widgets/multiline_text.hpp]: Updated comment.

**Comment by stevecotton** [comment]
Looks good to me. Can't approve in the Web GUI as it's my own PR, even though it's no longer my own code.

**Comment by babaissarkar** [comment]
> Looks good to me. Can't approve in the Web GUI as it's my own PR, even though it's no longer my own code.

What's the procedure in this case? Maybe somebody else should be merging this. (Or can I?)

**Comment by stevecotton** [comment]
I can hit the "Squash and merge" button, just can't mark it as approved. I'm currently waiting for the CI to finish.

**Review by babaissarkar — APPROVED** [review]

**Comment by babaissarkar** [comment]
I've marked it, hopefully that's not a problem.

**Comment by stevecotton** [comment]
Okay, when Github squashed it it only let me choose myself as the author, and credited you as co-author.

**Pull request merged**
